### ISSUE TITLE
Generate SHACL shapes 

### ISSUE THREAD

**Issue by vemonet** [linkml/linkml#316]
**Is your feature request related to a problem? Please describe.**
Generate SHACL shapes from the linkml YAML model

SHACL is the standard accepted by the W3C, it now starts to be integrated in multiple projects to perform data validation (e.g. Ontotext GraphDB triplestore, RDF4J).

Unfortunately ShEx shapes cannot be converted to SHACL shapes, and there are no work to integrate ShEx validation directly in existing products, so the use of ShEx to validate data is limited at running uncomplete third-party softwares

**Describe the solution you'd like**
The same way as ShEx shapes are generated it would be ideal if SHACL shapes could be generated from the linkml YAML model

**How important is this feature?** Select from the options below:
• Important - it's a blocker and can't do work without it

**When will use cases depending on this become relevant?** Select from the options below:
• Long-term - 6 months - 1 year

**Comment by cmungall** [comment]
Thanks @vemonet! Is this something you'd be able to contribute to - if not coding, providing test sets, evaluating beta releases...?

**Comment by vemonet** [comment]
I'll be interested to take a look into the code when I have the time, we should be able to reuse the ShEx generator and build the SHACL shapes with rdflib relatively easily: https://github.com/linkml/linkml/blob/main/linkml/generators/shexgen.py

For the test sets I was thinking reusing some snippets from the docs (https://biolink.github.io/biolink-model/about/mapping-rdf.html) and bringing in some nanopublications we made, but would be interested if you have other good cases

**Comment by cmungall** [comment]
@vemonet forgot to report back we have a beta version: https://linkml.io/linkml/generators/shacl.html

currently booleans and some other parts of shacl are not yet represented

**Comment by vemonet** [comment]
@cmungall I also forgot to report back! 

A few days ago I took a look into LinkML, to generate a model for the SemanticScience ontology: https://github.com/vemonet/semanticscience-model

I noticed you already implemented SHACL so I tried it, the SHACL shape generated seems like a really good start! I just need to figure out how to properly populate the slot from the OWL ontology, but that's another problem...

I added the prefixes of the model to make the SHACL more readable: https://github.com/linkml/linkml/pull/499

I also started a really basic generator for RML mappings (to generate mappings template to convert structured data to the model): https://github.com/vemonet/linkml/blob/add-gen-rml/linkml/generators/rmlgen.py
Is that something that would be interesting to add to LinkML when it will be more polished? I think the RML community would be interested to have such generator to have mappings to get started
